src/TraceEvent/Stacks/CallTree.cs

```diff
@@ -297,8 +297,11 @@ private CallTreeNode FindTreeNode(StackSourceCallStackIndex stack, RecursionGuar
         {
             if (recursionGuard.RequiresNewThread)
             {
+                var capturedThis = this;
```

Comment:
📝 I was surprised to find this explicit capture was required in order to avoid a closure allocation for the method.

/cc @jaredpar @agocke

Reply:
Looks right to me. `this` is a parameter, scope is the start of the method.